XamlAnimatedGif/Animator.cs

```diff
@@ -310,13 +370,12 @@ private async Task RenderFrameAsync(int frameIndex, CancellationToken cancellati
                 var palette = _palettes[frameIndex];
                 int transparencyIndex = palette.TransparencyIndex ?? -1;
 
-                var rows = desc.Interlace
-                    ? InterlacedRows(rect.Height)
-                    : NormalRows(rect.Height);
-
                 foreach (int y in rows)
                 {
-                    indexStream.ReadAll(indexBuffer, 0, desc.Width);
+                    if (!_cacheFrameDataInMemory)
+                    {
+                        indexStream.ReadAll(indexBuffer, 0, desc.Width);
+                    }
```

Comment:
If `_cacheFrameDataInMemory` is true, you could set `indexBuffer` to `indexDecompressedBytes`. This would avoid an additional test below when getting the index for `x`.

Reply:
now I only check for _cacheFrameDataInMemory once and populate indexBuffer accordingly